waffle-jest/test/matchers/calledOnContract/calledOnContractWith.test.ts

```diff
@@ -0,0 +1,85 @@
+import {MockProvider} from '@ethereum-waffle/provider';
+import {ContractFactory} from 'ethers';
+import {CALLS_ABI, CALLS_BYTECODE} from '../../contracts/Calls';
+
+async function setup() {
```

Comment:
Deploying the same contract for each test can be done in a fixture, which will speed the tests up significantly. This applies for each of the setup functions that you have that deploy contracts.

Reply:
Jest runs its test suites in parallel, and a custom test environment cannot share a stateful instance that is passed around (see [here](https://github.com/facebook/jest/issues/7184#issuecomment-616037304)).

The only way around this is if you run this mock before you even run your tests and have your tests call out to it.

Reply:
Let's keep it the way it is.